homeassistant/components/systemmonitor/sensor.py

```diff
@@ -60,62 +66,188 @@
 
 SIGNAL_SYSTEMMONITOR_UPDATE = "systemmonitor_update"
 
-# Schema: [name, unit of measurement, icon, device class, flag if mandatory arg]
-SENSOR_TYPES: dict[str, tuple[str, str | None, str | None, str | None, bool]] = {
-    "disk_free": ("Disk free", DATA_GIBIBYTES, "mdi:harddisk", None, False),
-    "disk_use": ("Disk use", DATA_GIBIBYTES, "mdi:harddisk", None, False),
-    "disk_use_percent": (
-        "Disk use (percent)",
-        PERCENTAGE,
-        "mdi:harddisk",
-        None,
-        False,
+
+@dataclass
+class SysMonitorSensorEntityDescription(SensorEntityDescription):
+    """Description for System Monitor sensor entities."""
+
+    mandatory_arg: bool = False
+
+
+SENSOR_TYPES: dict[str, SysMonitorSensorEntityDescription] = {
+    "disk_free": SysMonitorSensorEntityDescription(
+        key="disk_free",
+        name="Disk free",
+        native_unit_of_measurement=DATA_GIBIBYTES,
+        icon="mdi:harddisk",
+        state_class=STATE_CLASS_TOTAL,
+    ),
+    "disk_use": SysMonitorSensorEntityDescription(
+        key="disk_use",
+        name="Disk use",
+        native_unit_of_measurement=DATA_GIBIBYTES,
+        icon="mdi:harddisk",
+        state_class=STATE_CLASS_TOTAL,
+    ),
+    "disk_use_percent": SysMonitorSensorEntityDescription(
+        key="disk_use_percent",
+        name="Disk use (percent)",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:harddisk",
+        state_class=STATE_CLASS_TOTAL,
+    ),
+    "ipv4_address": SysMonitorSensorEntityDescription(
+        key="ipv4_address",
+        name="IPv4 address",
+        native_unit_of_measurement="",
```

Comment:
why define it?

Reply:
honestly, I have converted the prior `SENSOR_TYPES` dict by a regex search and replace and only removed everything which matched the default values, those this got missed 🙈